lighthouse-core/audits/installable-manifest.js

```diff
@@ -51,59 +125,108 @@ class InstallableManifest extends MultiCheckAudit {
   }
 
   /**
-   * @param {LH.Artifacts.ManifestValues} manifestValues
-   * @return {Array<string>}
+   * @param {LH.Artifacts} artifacts
+   * @return {{i18nErrors: Array<LH.IcuMessage | string>; warnings: Array<LH.IcuMessage>}}
    */
-  static assessManifest(manifestValues) {
-    if (manifestValues.isParseFailure && manifestValues.parseFailureReason) {
-      return [manifestValues.parseFailureReason];
+  static getInstallabilityErrors(artifacts) {
+    const installabilityErrors = artifacts.InstallabilityErrors.errors;
+    const i18nErrors = [];
+    const warnings = [];
+    const errorArgumentsRegex = /{([^}]+)}/g;
+
+    for (const err of installabilityErrors) {
+      // Filter out errorId 'in-incognito' since Lighthouse recommends incognito.
+      if (err.errorId === 'in-incognito') continue;
+
+      if (err.errorId === 'warn-not-offline-capable') {
+        warnings.push(str_(UIStrings[err.errorId]));
+        continue;
+      }
+
+      // @ts-expect-error errorIds from protocol should match up against the strings dict
+      const matchingString = UIStrings[err.errorId];
+
+      if (matchingString === undefined) {
+        i18nErrors.push(str_(UIStrings.noErrorId, {errorId: err.errorId}));
+        continue;
+      }
+
+      // Get the arguments of the installability error message.
```

Comment:
```suggestion
      // Get the i18n argument names of the installability error message, if any.
```
just a suggestion, but I wasn't sure what this meant the first time I read this. There may be better ways to clarify.